### DISCUSSION TITLE
How to use Source Assets as an input to graph-backed assets

### DISCUSSION THREAD

**Discussion by seanlindo** [dagster-io/dagster#17708] [Q&A] [answered]
### What's the issue or suggestion?

It's not clear how, or if possible, other ways to provide Source Assets as inputs. For instance, what if one wanted to utilize a source asset in one of the ops comprising a graph-backed asset?

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment, marked as answer]
Hey @seanlindo - here's an example of using a source asset as input to a graph-backed asset:

```python
from dagster import SourceAsset, graph_asset, op

my_source_asset = SourceAsset("my_source_asset")


@op
def my_op(input1):
    ...


@graph_asset
def my_graph_asset(my_source_asset):
    return my_op(my_source_asset)
```

Note that I changed the title to make it more specific, which I think will help others find this question and answer when they have the same one. If there are other situations where you're looking to understand how to use source assets as input, mind opening a separate discussion?
